File: 0111. 二叉树的最小深度/递归.cpp
// 16ms
class Solution {
public:
    int minDepth(TreeNode* root) {
        if (root == NULL) {
            return 0;
        } else if (root->left == NULL && root->right == NULL) {
            return 1;
        } else {
            int result = INT_MAX;
            if (root->left != NULL) {
                result = min(minDepth(root->left) + 1, result);
            }
            if (root->right != NULL) {
                result = min(minDepth(root->right) + 1, result);
            }
            return result;
        }
    }
};

File: 0145. 二叉树的后序遍历/递归.js
// 84ms
var postorderTraversal = function (root) {
    var result = []
    if (root != null) {
        result.push(postorderTraversal(root.left))
        result.push(postorderTraversal(root.right))
        result.push(root.val)
    }
    return result.flat()
};